### PR TITLE
feat(annotations): Disable annotations for excel and iWork formats

### PR DESCRIPTION
**Pull request by mxiao6**

### CODE REVIEW

File: src/lib/extensions.js [hunk 1]
```diff
@@ -48,8 +48,14 @@ export const CODE_EXTENSIONS = [
     'yaml',
 ];
 
+export const EXCEL_EXTENSIONS = ['xls', 'xlsm', 'xlsx'];
```
Comment (jstoffan) [inline src/lib/extensions.js:51]: Do we want to include `xlsb`? Should we update OfficeViewer to use parts of these constants?

File: src/lib/viewers/BaseViewer.js [hunk 2]
```diff
@@ -173,6 +174,10 @@ class BaseViewer extends EventEmitter {
             const fileExt = this.options.file.extension;
             this.fileLoadingIcon = getIconFromExtension(fileExt);
             this.startAt = getProp(this.options, `fileOptions.${this.options.file.id}.${FILE_OPTION_START}`, {});
+
+            if (EXCEL_EXTENSIONS.includes(fileExt) || IWORK_EXTENSIONS.includes(fileExt)) {
```
Comment (jstoffan) [inline src/lib/viewers/BaseViewer.js:178]: Seems like this logic should live in the `areAnnotationsEnabled` method.

File: src/lib/extensions.js (hunk 1 above)
Comment (mxiao6) [reply src/lib/extensions.js:51]: > Do we want to include `xlsb`? Should we update OfficeViewer to use parts of these constants?

`xlsb` conversion to pdf is not supported. But OfficeViewer supports `xlsb` but not `xls`...

File: src/lib/viewers/BaseViewer.js [hunk 3]
```diff
@@ -1030,6 +1031,11 @@ class BaseViewer extends EventEmitter {
             return this.viewerConfig.enabled;
         }
 
+        // Disable new annotations for Excel and iWork formats
+        if (EXCEL_EXTENSIONS.includes(extension) || IWORK_EXTENSIONS.includes(extension)) {
+            this.options.showAnnotationsControls = false;
```
Comment (jstoffan) [inline src/lib/viewers/BaseViewer.js:1036]: Do we want to mutate the options object in here? Seems like we should just `return false`. Do we need a new method for the new experience?

File: src/lib/extensions.js (hunk 1 above)
Comment (ConradJChan) [reply src/lib/extensions.js:51]: Should we add a comment so this nuance isn't lost?

File: src/lib/viewers/__tests__/BaseViewer-test.js [hunk 4]
```diff
@@ -1254,6 +1254,27 @@ describe('lib/viewers/BaseViewer', () => {
         });
     });
 
+    describe('areNewAnnotationsEnabled()', () => {
+        it('should return false if the file is excel or iWork formats', () => {
```
Comment (ConradJChan) [inline src/lib/viewers/__tests__/BaseViewer-test.js:1258]: Should we have a parameterized test that runs over all the excluded extensions? Might make it easier to assert any changes in the future

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 5]
```diff
@@ -1015,7 +1015,7 @@ class DocBaseViewer extends BaseViewer {
         this.controls = new Controls(this.containerEl);
         this.pageControls = new PageControls(this.controls, this.docEl);
         this.zoomControls = new ZoomControls(this.controls);
-        if (this.options.showAnnotationsControls) {
+        if (this.areNewAnnotationsEnabled()) {
             this.annotationControls = new AnnotationControls(this.controls);
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocBaseViewer.js:1019]: This disables the controls, but doesn't disable the Annotations SDK, so annotations that great created in a different way (platform, API, etc.) will still be shown. Are we concerned about that scenario?

Comment (mxiao6) [reply src/lib/viewers/doc/DocBaseViewer.js:1019]: > This disables the controls, but doesn't disable the Annotations SDK, so annotations that great created in a different way (platform, API, etc.) will still be shown. Are we concerned about that scenario?

Yeah, Rohit said we only want to disable new annotations for these extensions.

Comment (jstoffan) [reply src/lib/viewers/doc/DocBaseViewer.js:1019]: Does that mean "newly-created" or "new experience"? I suspect that he meant the former. Can you check?

Comment (mxiao6) [reply src/lib/viewers/doc/DocBaseViewer.js:1019]: > Does that mean "newly-created" or "new experience"? I suspect that he meant the latter. Can you check?

Oh I realized this difference just now...

**Pull request merged**